src/components/containers/SubplotAccordion.js

```diff
@@ -0,0 +1,129 @@
+import PlotlyFold from './PlotlyFold';
+import TraceRequiredPanel from './TraceRequiredPanel';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {
+  connectTraceToPlot,
+  connectCartesianSubplotToLayout,
+  connectNonCartesianSubplotToLayout,
+} from 'lib';
+import {TRACE_TO_AXIS, AXIS_TO_ATTR} from 'lib/constants';
+
+const TraceFold = connectTraceToPlot(PlotlyFold);
+const NonCartesianSubplotFold = connectNonCartesianSubplotToLayout(PlotlyFold);
+const CartesianSubplotFold = connectCartesianSubplotToLayout(PlotlyFold);
+
+class SubplotAccordion extends Component {
+  render() {
+    const {data = [], layout = {}} = this.context;
+    const {children, messageIfEmptyFold} = this.props;
+    const subplotFolds = [];
+
+    const allCartesianAxisCombinations = data.reduce((acc, curVal, inx) => {
+      if (TRACE_TO_AXIS.cartesian.some(c => c === curVal.type)) {
+        const xaxis = 'xaxis' + (curVal.xaxis ? curVal.xaxis.substring(1) : '');
+        const yaxis = 'yaxis' + (curVal.yaxis ? curVal.yaxis.substring(1) : '');
+
+        const existingComboIndex = acc.findIndex(
+          t => t.xaxis === xaxis && t.yaxis === yaxis
+        );
+        if (existingComboIndex === -1) {
+          acc.push({
+            xaxis: xaxis,
+            yaxis: yaxis,
+            xaxisName: curVal.xaxis || 'x1',
+            yaxisName: curVal.yaxis || 'y1',
+            index: [inx],
+          });
+        } else {
+          acc[existingComboIndex].index.push(inx);
+        }
+      }
+      return acc;
+    }, []);
+
+    allCartesianAxisCombinations.forEach(
+      d =>
+        (subplotFolds[d.index[0]] = (
+          <CartesianSubplotFold
+            key={d.index[0]}
+            traceIndexes={d.index}
+            canDelete={false}
+            messageIfEmpty={messageIfEmptyFold}
+            xaxis={d.xaxis}
+            yaxis={d.yaxis}
+            name={`${d.xaxisName} ${d.yaxisName}`}
+          >
+            {children}
+          </CartesianSubplotFold>
+        ))
+    );
+
+    Object.keys(layout).forEach(layoutKey => {
```

Comment:
I found it confusing here as to why we were looping through layout, not fullLayout.
And why we were using data.findIndex instead of maybe filter to find all indexes that could meet the condition.

If you could maybe comment above on those 2 questions, I think it may help in a few months time, when revisiting that code.